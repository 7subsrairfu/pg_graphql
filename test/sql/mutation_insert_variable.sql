begin;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );

    select graphql.resolve($$
    mutation createAccount($emailAddress: String) {
       xyz: insertAccount(object: {
        email: $emailAddress
      }) {
        id
      }
    }
    $$,
    variables := '{"emailAddress": "foo@bar.com"}'::jsonb
    );

rollback;
