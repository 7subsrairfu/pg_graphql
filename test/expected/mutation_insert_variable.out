begin;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    select graphql.resolve($$
    mutation createAccount($emailAddress: String) {
       xyz: insertAccount(object: {
        email: $emailAddress
      }) {
        id
      }
    }
    $$,
    variables := '{"emailAddress": "foo@bar.com"}'::jsonb
    );
                  resolve                   
--------------------------------------------
 {"data": {"xyz": {"id": 1}}, "errors": []}
(1 row)

rollback;
